### PR TITLE
QWen convert docs

### PR DESCRIPTION
**Pull request by xuxinyi389**
### PR types
Docs
### PR changes
Docs
### Description
Pcard-73263
补充qwen的映射文档。

### CODE REVIEW

File: docs/guides/model_convert/convert_from_others/flash_attn/flash_attn.flash_attn_interface.flash_attn_func.md [hunk 1]
```diff
@@ -0,0 +1,34 @@
+## [torch 参数更多]flash_attn.flash_attn_interface.flash_attn_func
+
+### [flash_attn.flash_attn_interface.flash_attn_func](https://github.com/Dao-AILab/flash-attention/blob/72e27c6320555a37a83338178caa25a388e46121/flash_attn/flash_attn_interface.py#L808)
+
+```python
+flash_attn.flash_attn_interface.flash_attn_func(q,k,v,dropout_p=0.0,softmax_scale=None,causal=False,window_size=(-1, -1),softcap=0.0,alibi_slopes=None,deterministic=False,return_attn_probs=False)
```
Comment (zhwesky2010) [inline docs/guides/model_convert/convert_from_others/flash_attn/flash_attn.flash_attn_interface.flash_attn_func.md]: 建议写成python的规范风格：参数与参数之间空一格，不然挤在一起看起来风格比较差

infoflow 2024-07-17 19-24-37

File: docs/guides/model_convert/convert_from_others/flash_attn/flash_attn.flash_attn_interface.flash_attn_func.md [hunk 2]
```diff
@@ -0,0 +1,34 @@
...
+```
+
+### [paddle.nn.functional.flash_attention.flash_attention](https://github.com/PaddlePaddle/Paddle/blob/900d27c40ef4567d7ea6342f3f0eedd394885ecb/python/paddle/nn/functional/flash_attention.py#L248)
+
+```python
+paddle.nn.functional.flash_attention.flash_attention(query,key,value,dropout=0.0,causal=False，return_softmax=False,*,fixed_seed_offset=None,rng_name="",training=True)
```
Comment (zhwesky2010) [inline docs/guides/model_convert/convert_from_others/flash_attn/flash_attn.flash_attn_interface.flash_attn_func.md]: 风格参考原API文档，形式规范一点

File: docs/guides/model_convert/convert_from_others/flash_attn/flash_attn.flash_attn_interface.flash_attn_unpadded_func.md [hunk 3]
```diff
@@ -0,0 +1,41 @@
...
+```python
+paddle.nn.functional.flash_attention.flash_attn_unpadded(query,key,value,cu_seqlens_q,cu_seqlens_k,
+max_seqlen_q,max_seqlen_k,scale,dropout=0.0,causal=False,return_softmax=False,fixed_seed_offset=None,rng_name='',training=True,name=None)
+```
+
+其中 PyTorch 相比 Paddle 支持更多其他参数，具体如下：
+
+### 参数映射
+
+| PyTorch           | PaddlePaddle      | 备注                |
```
Comment (zhwesky2010) [inline docs/guides/model_convert/convert_from_others/flash_attn/flash_attn.flash_attn_interface.flash_attn_unpadded_func.md]: 格式：
infoflow 2024-07-17 19-28-45

备注里没有空着的情况

File: docs/guides/model_convert/convert_from_others/flash_attn/flash_attn.layers.rotary.apply_rotary_emb_func.md [hunk 4]
```diff
@@ -0,0 +1,22 @@
...
+
+### [flash_attn.layers.rotary.apply_rotary_emb_fun](https://github.com/Dao-AILab/flash-attention/blob/d0787acc16c3667156b51ce5b01bdafc7594ed39/flash_attn/layers/rotary.py#L94)
+
+```python
+flash_attn.layers.rotary.apply_rotary_emb_fun(x,cos,sin,interleaved=False,inplace=False,seqlen_offsets: Union[int, torch.Tensor] = 0,cu_seqlens: Optional[torch.Tensor] = None,max_seqlen: Optional[int] = None,)
+```
+
+Paddle 无此 API，需要组合实现。
+
+### 参数映射
```
Comment (zhwesky2010) [inline docs/guides/model_convert/convert_from_others/flash_attn/flash_attn.layers.rotary.apply_rotary_emb_func.md]: 格式：

infoflow 2024-07-17 19-30-18


组合替代实现没有参数映射，这个是没有意义的

File: docs/guides/model_convert/convert_from_others/flash_attn/flash_attn.ops.rms_norm.rms_norm.md [hunk 5]
```diff
@@ -0,0 +1,31 @@
...
+其中 Paddle 相比 PyTorch 支持更多其他参数，具体如下：
+
+### 参数映射
+
+| PyTorch           | PaddlePaddle      | 备注                |
+| ----------------- | ----------------- | ------------------------------------------------------------ |
+| x                 | x                 |  |
+| weight            | norm_weight       |  |
+| epsilon           | epsilon           |  |
+| -                 | norm_bias         |  用于仿射输出的偏置张量 |
```
Comment (zhwesky2010) [inline docs/guides/model_convert/convert_from_others/flash_attn/flash_attn.ops.rms_norm.rms_norm.md]: 这些都不符合规范，备注没写，paddle多的参数如何设置也没写

File: docs/guides/model_convert/convert_from_others/transformers/transformers.GenerationConfig.md [hunk 6]
```diff
@@ -0,0 +1,15 @@
...
+transformers.GenerationConfig(*kwargs)
+```
+
+### [paddlenlp.generation.GenerationConfig](https://github.com/PaddlePaddle/PaddleNLP/blob/e336e78c338d2514ee6c937982ce5d8c960b85ff/paddlenlp/generation/configuration_utils.py#L62)
+
+```python
+paddlenlp.generation.GenerationConfig(*kwargs)
+```
+
+两者功能一致但参数不一致，torch 的参数更多。参数数量众多，详情可参考 PaConvert 工具。
```
Comment (zhwesky2010) [inline docs/guides/model_convert/convert_from_others/transformers/transformers.GenerationConfig.md]: 这个不能这么写

File: docs/guides/model_convert/convert_from_others/transformers/transformers.PreTrainedTokenizer.md [hunk 7]
```diff
@@ -0,0 +1,15 @@
+## [参数完全一致]transformers.PreTrainedTokenizer
```
Comment (zhwesky2010) [inline docs/guides/model_convert/convert_from_others/transformers/transformers.PreTrainedTokenizer.md]: 这些和PaConvert是不是一致的，如果有功能不一致的，就需要写出来

File: docs/guides/model_convert/convert_from_others/transformers/transformers.PreTrainedModel.md [hunk 8]
```diff
@@ -0,0 +1,15 @@
+## [参数完全一致]transformers.PreTrainedModel
```
Comment (zhwesky2010) [inline docs/guides/model_convert/convert_from_others/transformers/transformers.PreTrainedModel.md]: 这些和PaConvert的规则是不是一致的，如果有功能不一致的，就需要写出来

File: docs/guides/model_convert/convert_from_others/transformers/transformers.PretrainedConfig.md [hunk 9]
```diff
@@ -0,0 +1,15 @@
...
+transformers.PretrainedConfig(*kwargs)
+```
+
+### [paddlenlp.transformers.PretrainedConfig](https://github.com/PaddlePaddle/PaddleNLP/blob/e336e78c338d2514ee6c937982ce5d8c960b85ff/paddlenlp/generation/configuration_utils.py#L62)
+
+```python
+paddlenlp.transformers.PretrainedConfig(*kwargs)
+```
+
+两者功能一致但参数不一致，torch 的参数更多。参数数量众多，详情可参考 PaConvert 工具。
```
Comment (zhwesky2010) [inline docs/guides/model_convert/convert_from_others/transformers/transformers.PretrainedConfig.md]: 不能这么写

File: docs/guides/model_convert/convert_from_others/transformers/transformers.StoppingCriteriaList.md [hunk 10]
```diff
@@ -0,0 +1,22 @@
...
+transformers.StoppingCriteriaList(input_ids: torch.LongTensor, scores: torch.FloatTensor, **kwargs)
+```
+
+### [paddlenlp.generation.StoppingCriteriaList](https://github.com/PaddlePaddle/PaddleNLP/blob/e336e78c338d2514ee6c937982ce5d8c960b85ff/paddlenlp/generation/stopping_criteria.py#L72)
+
+```python
+paddlenlp.generation.StoppingCriteriaList(input_ids: paddle.Tensor, scores: paddle.Tensor, **kwargs)
+```
+
+两者功能一致但参数不一致，部分参数名不同，具体如下：
```
Comment (zhwesky2010) [inline docs/guides/model_convert/convert_from_others/transformers/transformers.StoppingCriteriaList.md]: 这个参数映射里 什么也看不到

**Review by zhwesky2010 — COMMENTED** [review]
没有review完，内容和格式还有较多需要修改的。写完可以自己先通读自查一遍。

1. 注意内容：文档和PaConvert Matcher的思路需要对齐，不要出现转写思路上的Diff，写法上倒可以灵活无需生硬，因为一个是给人看到，一个是给机器看的。
2. 注意格式规范。

这个应该还没有加映射首页的表格

File: docs/guides/model_convert/convert_from_pytorch/api_difference_third_party/flash_attn/flash_attn.ops.rms_norm.rms_norm.md [hunk 11]
```diff
@@ -0,0 +1,31 @@
+## [仅 paddle 参数更多]flash_attn.ops.rms_norm.rms_norm
```
Comment (zhwesky2010) [inline docs/guides/model_convert/convert_from_pytorch/api_difference_third_party/flash_attn/flash_attn.ops.rms_norm.rms_norm.md]: https://github.com/PaddlePaddle/docs/blob/develop/docs/guides/model_convert/convert_from_pytorch/api_difference/pytorch_api_mapping_format_cn.md

现在换成了 paddle参数更多

File: docs/guides/model_convert/convert_from_pytorch/api_difference_third_party/transformers/transformers.PreTrainedModel.generate.md [hunk 12]
```diff
@@ -0,0 +1,31 @@
...
+```python
+paddlenlp.transformers.PreTrainedTokenizer.encode(input_ids, **kwargs)
+```
+
+### 参数映射
+
+| PyTorch       | PaddlePaddle | 备注                                                   |
+| ------------- | ------------ | ------------------------------------------------------ |
+| input         | input_ids    | token 的 id 组成的 Tensor。 |
+| 返回值         | 返回值        | PyTorch 返回的 Tensor 包含输入的 Tensor，Paddle 不包含且返回类型为 tuple。|
```
Comment (zhwesky2010) [inline docs/guides/model_convert/convert_from_pytorch/api_difference_third_party/transformers/transformers.PreTrainedModel.generate.md]: 这里就写清楚：Pytorch返回什么，Paddle返回什么
这里写包含不包含，看得不太清晰

File: docs/guides/model_convert/convert_from_pytorch/api_difference_third_party/transformers/transformers.PreTrainedModel.get_head_mask.md [hunk 13]
```diff
@@ -0,0 +1,30 @@
...
+transformers.PreTrainedModel.get_head_mask(head_mask: Optional, num_hidden_layers: int, is_attention_chunked: bool = False)
+```
+
+Paddle 无此 API，需要组合实现。
+
+### 转写示例
+
+```python
+# Pytorch 写法
+head_mask_output = transformers.logging.get_logger(head_mask = x, num_hidden_layers, is_attention_chunked)
```
Comment (zhwesky2010) [inline docs/guides/model_convert/convert_from_pytorch/api_difference_third_party/transformers/transformers.PreTrainedModel.get_head_mask.md]: 这个不是在转写get_head_mask？

File: docs/guides/model_convert/convert_from_pytorch/api_difference_third_party/transformers/transformers.GenerationConfig.md [hunk 14]
```diff
@@ -0,0 +1,68 @@
...
+paddlenlp.generation.GenerationConfig(*kwargs)
+```
+
+两者功能一致，但 PyTorch 相比 Paddle 支持更多其他参数，具体如下：
+
+### 参数映射
+
+| transformers                         | PaddlePaddle        | 备注      |
+| -------------------------------------| ------------------- | -------- |
+| max_length                           | max_length          | 最大生成长度。 |
```
Comment (zhwesky2010) [inline docs/guides/model_convert/convert_from_pytorch/api_difference_third_party/transformers/transformers.GenerationConfig.md:21]: 这些和PaConvert对应上了吧：
- 可直接删除对应的是："a": ""
- 暂无转写方式对应的是unsupport_args

Paconvert中好像没这么多unsupport_args?

File: docs/guides/model_convert/convert_from_pytorch/api_difference_third_party/transformers/transformers.PreTrainedTokenizer.encode.md [hunk 15]
```diff
@@ -0,0 +1,30 @@
...
+```python
+paddlenlp.transformers.PreTrainedTokenizer.encode(text, **kwargs)
+```
+
+### 参数映射
+
+| PyTorch       | PaddlePaddle | 备注                                                   |
+| ------------- | ------------ | ------------------------------------------------------ |
+| text          | text         | 输入的编码文本。  |
+| 返回值         | 返回值        | PyTorch 返回 Tensor 类型，Paddle 返回一个字典类型。|
```
Comment (zhwesky2010) [inline docs/guides/model_convert/convert_from_pytorch/api_difference_third_party/transformers/transformers.PreTrainedTokenizer.encode.md]: 可详细一点：字典中包含多个Tensor？后面再加一句：需要转写

File: docs/guides/model_convert/convert_from_pytorch/api_difference_third_party/transformers/transformers.StoppingCriteriaList.md [hunk 16]
```diff
@@ -0,0 +1,22 @@
...
+```
+
+两者功能一致但参数不一致，部分参数名不同，具体如下：
+
+### 参数映射
+
+| PyTorch           | PaddlePaddle      | 备注                                     |
+| ----------------- | ----------------- | --------------------------------------- |
+| input_ids         | input_ids         | 输入(tokens) 的 id 组成的 Tensor。 |
+| scores            | logits            | 得分组成的 Tensor。 |
```
Comment (zhwesky2010) [inline docs/guides/model_convert/convert_from_pytorch/api_difference_third_party/transformers/transformers.StoppingCriteriaList.md]: 仅参数名不同？

这个没有按模板来写，用法不一致也需要具体写出来

File: docs/guides/model_convert/convert_from_pytorch/api_difference_third_party/transformers/transformers.generation.LogitsProcessor.md [hunk 17]
```diff
@@ -0,0 +1,22 @@
...
+```
+
+两者功能一致但参数名不一致，部分参数名不同，具体如
+
+### 参数映射
+
+| PyTorch           | PaddlePaddle      | 备注                                     |
+| ----------------- | ----------------- | --------------------------------------- |
+| input_ids         | input_ids         | 输入(tokens) 的 id 组成的 Tensor。 |
+| scores            | logits            | 得分组成的 Tensor。 |
```
Comment (zhwesky2010) [inline docs/guides/model_convert/convert_from_pytorch/api_difference_third_party/transformers/transformers.generation.LogitsProcessor.md]: 后面需要加一句：仅参数名不一致


https://github.com/PaddlePaddle/docs/blob/develop/docs/guides/model_convert/convert_from_pytorch/api_difference/pytorch_api_mapping_format_cn.md 模板和规范都重新更新了，可再阅读一遍

File: docs/guides/model_convert/convert_from_pytorch/api_difference_third_party/transformers/transformers.logging.get_logger.md [hunk 18]
```diff
@@ -0,0 +1,19 @@
...
+Paddle 无此 API，需要组合实现。
+
+### 转写示例
+
+```python
+# Pytorch 写法
+transformers.logging.get_logger()
+
+# Paddle 写法
+paddle.utils.try_import('logging').getLogger(name=__name__)
```
Comment (zhwesky2010) [inline docs/guides/model_convert/convert_from_pytorch/api_difference_third_party/transformers/transformers.logging.get_logger.md]:
```
import logging
logging.getgetLogger()
```

File: docs/guides/model_convert/convert_from_pytorch/api_difference_third_party/transformers/transformers.GenerationConfig.md (hunk 14 above)
Comment (xuxinyi389) [reply docs/guides/model_convert/convert_from_pytorch/api_difference_third_party/transformers/transformers.GenerationConfig.md:21]: Paconvert 中的 unsupport args 会更多一些，将文档中标明的可直接删除的args也包含了，后面修改下 PaConvert中的 api-mapping.json即可

**Review by zhwesky2010 — APPROVED** [review]
